src/main/java/com/jmb/springfactory/service/task/TaskServiceImpl.java

```diff
@@ -29,122 +34,185 @@
+     * @throws PersistenceLayerException
+     */
+    private Integer getLastPositionOrder(Integer orderId) throws PersistenceLayerException {
+
+        final Integer numberOrders = taskMySQLService.countByOrderId(orderId).intValue();
+        final Integer lastNumberOrder = numberOrders + 1;
+
+        serviceLog.info(String.format("There are %s task for order %s, the order of the new one is %s", orderId,
+                numberOrders, lastNumberOrder));
+
+        return lastNumberOrder;
+    }
+
+    @Override
+    public List<TaskDto> findAll(QueryTaskObject queryParams) {
+        serviceLog.info(String.format("Search task with the follow arguments: %s", queryParams.toString()));
+        return taskMySQLService.findAll(queryParams).map(this::entityToDto).collect(Collectors.toList());
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void addComment(Integer idTask, CommentDto comment) throws NotFoundException, ServiceLayerException {
+
+        final TaskDto task = findOne(idTask);
+        task.setComments((List<CommentDto>) addIntoList(task.getComments(), comment));
+        save(task);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void addWorkLog(Integer idTask, WorkLogDto workLog) throws NotFoundException, ServiceLayerException {
+
+        final TaskDto task = findOne(idTask);
+        task.setWorkLogs((List<WorkLogDto>) addIntoList(task.getWorkLogs(), workLog));
+        save(task);
+    }
+
+    @Override
+    public Optional<TaskDto> findOneById(Integer id) {
+        return taskMySQLService.findOne(id).map(this::entityToDto);
+    }
+
+    @Override
+    public Task merge(TaskDto dto, Task entity) {
+
+        if (UtilsService.existAll(dto, entity)) {
+            mergeProductionOrder(dto, entity);
+            mergeWorkGroup(dto, entity);
+        }
+
+        return super.merge(dto, entity);
+    }
+
+    private void mergeWorkGroup(TaskDto dto, Task entity) {
+        Optional.ofNullable(entity).map(Task::getGroupAssigned).map(WorkGroup::getId).map(this::findGroupById)
+                .ifPresent(dto::setGroupAssigned);
+    }
+
+    private void mergeProductionOrder(TaskDto dto, Task entity) {
+        ProductionOrderDTO orderDto = null;
+        try {
+            if (UtilsService.exist(entity.getOrder())) {
+                orderDto = productionOrderService.findOne(entity.getOrder().getId());
+            }
+        } catch (NotFoundException e) {
+            serviceLog.warn(e.getMessage());
+        }
+        dto.setOrder(orderDto);
+    }
+
+    private WorkGroupDto findGroupById(Integer id) {
+        WorkGroupDto group = null;
+        try {
+            group = groupService.findOne(id);
+        } catch (NotFoundException e) {
+            serviceLog.error(e.getMessage());
+            group = null;
+        }
+        return group;
+    }
+
+    @Override
+    public TaskDto saveByGroup(TaskDto task, Integer groupId) throws NotFoundException, ServiceLayerException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)